### PR TITLE
:bug: exit after showing help

### LINKED ISSUE #439: [BUG] "konveyor-analyzer-dep --help" keeps running instead of showing help and stopping
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-beta2

### Priority

Minor

### Current Behavior

When running `konveyor-analyzer-dep --help`, the command shows the help and keeps running instead of exiting.

### Expected Behavior

The command should show the help and exit.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Run `[docker|podman] run -ti --entrypoint /bin/bash <latest analyzer-lsp image>`
2. Run `konveyor-analyzer-help` within the container.

### Environment

```markdown
- Fedora 38
- Latest analyzer-lsp image
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Vickysomtee**
Fixes #439

### CODE REVIEW

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
@Vickysomtee hey thank you for your fix. it should work! But I noticed that when we integrated cobra, we didn't do things quite the "cobra" way. Are you willing to help us refactor that code? The changes I am looking for are - 
1. main() function only calls cmd.Execute() The actual code that runs the analysis is removed from main()
2. analysis code goes into command's RunE() funcion instead 

Here's a good example of using different hooks in a cobra command: https://github.com/konveyor/kantra/blob/20af46fafa3876cd56f5b3be20610c7185673142/cmd/analyze.go#L85-L150

**Comment by Vickysomtee** [comment]
> @Vickysomtee hey thank you for your fix. it should work! But I noticed that when we integrated cobra, we didn't do things quite the "cobra" way. Are you willing to help us refactor that code? The changes I am looking for are -
> 
> 1. main() function only calls cmd.Execute() The actual code that runs the analysis is removed from main()
> 2. analysis code goes into command's RunE() funcion instead
> 
> Here's a good example of using different hooks in a cobra command: https://github.com/konveyor/kantra/blob/20af46fafa3876cd56f5b3be20610c7185673142/cmd/analyze.go#L85-L150

@pranavgaikwad Yes, I  would love to refactor this. Few questions though

1. Can I continue this refactor in this same PR?
2. Can you help highlight the code that handles the analysis so I can go through it and understand it better.
3. Are we writing tests for this? Additional context will be appreciated

**Comment by shawn-hurley** [comment]
> 1. Can I continue this refactor in this same PR?

Yeah absolutely

> 2. Can you help highlight the code that handles the analysis so I can go through it and understand it better.

https://github.com/konveyor/analyzer-lsp/blob/main/cmd/analyzer/main.go#L201

This is where the engine runs the rules, but all the setup is also very important

https://github.com/konveyor/analyzer-lsp/blob/main/cmd/analyzer/main.go#L96-L199

The important parts here are rule parsing and getting and setting up the correct providers.

> 3. Are we writing tests for this? Additional context will be appreciated.

No test will be needed for this refactor, our e2e tests will test this!

**Comment by Vickysomtee** [comment]
@shawn-hurley Thank you!

**Comment by Vickysomtee** [comment]
@shawn-hurley 

I think I might need help to get the demo test working but it's not currently the blocker for this PR. I had a little tight schedule but I am free now and I will continue on this. If I still need help on the demo test, I would let you know. Thank you!

**Comment by Vickysomtee** [comment]
I have updated the PR with the refactor @pranavgaikwad @shawn-hurley

**Comment by pranavgaikwad** [comment]
@Vickysomtee Thank you. Really appreciate you helping out. Is it ready for review?

File: cmd/analyzer/main.go [hunk 1]
```diff
@@ -87,142 +55,180 @@ func main() {
...
+	rootCmd.Flags().StringVar(&analysisMode, "analysis-mode", "", "select one of full or source-only to tell the providers what to analyize. This can be given on a per provider setting, but this flag will override")
+	rootCmd.Flags().BoolVar(&noDependencyRules, "no-dependency-rules", false, "Disable dependency analysis rules")
+	rootCmd.Flags().IntVar(&contextLines, "context-lines", 10, "When violation occurs, A part of source code is added to the output, So this flag configures the number of source code lines to be printed to the output.")
+
+	return rootCmd
+}
+
+func main() {
+	if err := AnalysisCmd().Execute(); err != nil {
+		println(err.Error())
```
Comment (pranavgaikwad) [inline cmd/analyzer/main.go:229]: Can we use `os.Exit(1)` here?

File: cmd/analyzer/main.go [hunk 2]
```diff
@@ -87,142 +55,180 @@ func main() {
...
-		errLog.Error(err, "failed to validate flags")
-		os.Exit(1)
-	}
-
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	defer cancelFunc()
+	rootCmd := &cobra.Command{
+		Use:   "analyze",
+		Short: "Tool for working with analyzer-lsp",
+		PreRun: func(c *cobra.Command, args []string) {
```
Comment (pranavgaikwad) [inline cmd/analyzer/main.go:61]: Instead of exiting here, can we use PreRunE function and only return an error?

**Review by pranavgaikwad — APPROVED** [review]

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
Requesting minor changes

File: cmd/analyzer/main.go [hunk 3]
```diff
@@ -87,142 +55,181 @@ func main() {
...
-		errLog.Error(err, "failed to validate flags")
-		os.Exit(1)
-	}
+	rootCmd := &cobra.Command{
+		Use:   "analyze",
+		Short: "Tool for working with analyzer-lsp",
+		PreRunE: func(c *cobra.Command, args []string) error {
+			err := validateFlags()
+			if err != nil {
+				errLog.Error(err, "failed to validate flags")
```
Comment (pranavgaikwad) [inline cmd/analyzer/main.go:64]: nit: can we `return err` here? and return nil after the if block? just making sure if we make changes in future, it doesn't go unnoticed that err could be non-nil after the if block

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
One small request

**Review by pranavgaikwad — APPROVED** [review]
Thank you @Vickysomtee

**Review by shawn-hurley — APPROVED** [review]
Thank you for the refactor! this is looking really good

**Pull request merged**